### DISCUSSION TITLE
When is the julia setup file executed?

### DISCUSSION THREAD

**Discussion by tpoisot** [quarto-dev/quarto-cli#7303] [Q&A] [answered]
### Description

The Julia setup file (https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/jupyter/lang/julia/setup.jl) is setting theme arguments for CairoMakie -- is there a place in the documentation that outlines *when* this file is executed?

If not, there is a risk of the user calling `CairoMakie.set_theme!()` early in the script and removing some of the default options, which would create some discrepancies between the results when using quarto and the results when using `update_theme!()`, especially when passing screen config arguments this way.

**Comment by cscheid** [comment, marked as answer]
(I agree this is a mess, and I know precisely why you're asking!)

It's an artificially-inserted first cell in the notebook. It happens in [line 127 of notebook.py](https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/jupyter/notebook.py#L127).

Just to keep you in the loop: we're currently having a discussion about how to expose and document this part to minimize the chance of users spending time chasing down this problem. Thanks for the patience!

**Reply by cscheid** [reply]
More concretely in the case of julia, [the code in `setup.jl` is interpolated with the appropriate values and then added as a code cell](https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/jupyter/lang/julia/setup.jl).

**Reply by tpoisot** [reply]
Nice - I started looking for possible issues after using `set_theme!` and  `activate!` gave strange results. Now that I've seen the code, I know what to avoid when changing the themes.
